app/src/main/java/fr/free/nrw/commons/mwapi/MediaWikiApi.java

```diff
@@ -105,4 +107,8 @@
     interface ProgressListener {
         void onProgress(long transferred, long total);
     }
+
+    @NonNull
+    String getWikiText(PageTitle title) throws IOException;
```

Comment:
We do not need a new API to check if the page has been nominated for deletion. 

Checkout `MediaDataExtractor`. It uses: 

```
        Single<Boolean> pageExistsSingle = mediaWikiApi.pageExists("Commons:Deletion_requests/" + filename);
```